### PR TITLE
docs(efs): README and doc string refresh

### PR DESCRIPTION
**Pull request by shivlaks**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-efs/README.md [hunk 1]
```diff
@@ -21,54 +21,78 @@
...
 
 Any access point that has been created outside the stack can be imported into your CDK app.
 
 Use the `fromAccessPointAttributes()` API to import an existing access point.
 
 ```ts
+import * as efs from '@aws-cdk/aws-efs';
+
 efs.AccessPoint.fromAccessPointAttributes(this, 'ap', {
   accessPointArn: 'fsap-1293c4d9832fo0912',
```
Comment (njlynch) [inline packages/@aws-cdk/aws-efs/README.md:97]:
```suggestion
  accessPointId: 'fsap-1293c4d9832fo0912',
```

File: packages/@aws-cdk/aws-efs/README.md [hunk 2]
```diff
@@ -109,7 +142,7 @@ const fileSystem = new efs.FileSystem(this, 'MyEfsFileSystem', {
   enableAutomaticBackups: true
 });
 
-const inst = new Instance(this, 'inst', {
+const instance = new ec2.Instance(this, 'inst', {
```
Comment (njlynch) [inline packages/@aws-cdk/aws-efs/README.md:145]: Need to rename the usage below to match.

File: packages/@aws-cdk/aws-efs/README.md [hunk 3]
```diff
@@ -109,7 +142,7 @@ const fileSystem = new efs.FileSystem(this, 'MyEfsFileSystem', {
...
   instanceType: InstanceType.of(InstanceClass.T2, InstanceSize.LARGE),
   machineImage: new AmazonLinuxImage({
```
Comment (njlynch) [inline packages/@aws-cdk/aws-efs/README.md:147]:
```suggestion
  machineImage: new ec2.AmazonLinuxImage({
```

File: packages/@aws-cdk/aws-efs/README.md [hunk 4]
```diff
@@ -109,7 +142,7 @@ const fileSystem = new efs.FileSystem(this, 'MyEfsFileSystem', {
   enableAutomaticBackups: true
 });
 
-const inst = new Instance(this, 'inst', {
+const instance = new ec2.Instance(this, 'inst', {
   instanceType: InstanceType.of(InstanceClass.T2, InstanceSize.LARGE),
```
Comment (njlynch) [inline packages/@aws-cdk/aws-efs/README.md:146]:
```suggestion
  instanceType: ec2.InstanceType.of(ec.InstanceClass.T2, ec2.InstanceSize.LARGE),
```

Several others throughout this example.

**Review by njlynch — COMMENTED** [review]
Some tweaks to the README

File: packages/@aws-cdk/aws-efs/README.md [hunk 5]
```diff
@@ -21,54 +21,78 @@
...
-  performanceMode: efs.PerformanceMode.GENERAL_PURPOSE,
-  throughputMode: efs.ThroughputMode.BURSTING
+  vpc: new ec2.Vpc(this, 'VPC'),
+  encrypted: true, // file system is not encrypted by default
+  lifecyclePolicy: efs.LifecyclePolicy.AFTER_14_DAYS, // files are not transitioned to infrequent access (IA) storage by default
 });
 ```
 
-A file system can set `RemovalPolicy`. Default policy is `RETAIN`.
+Since file systems are stateful resources, by default the file system will not be deleted when your
```
[the remark this answers is not in the document]
Comment (shivlaks) [reply packages/@aws-cdk/aws-efs/README.md]: moved it to a section called `deleting` and to the bottom

File: packages/@aws-cdk/aws-efs/README.md [hunk 6]
```diff
@@ -21,54 +21,78 @@
...
+
+## File Systems
+
+Amazon EFS provides elastic, shared file storage that is POSIX-compliant. The file system you create
+supports concurrent read and write access from multiple Amazon EC2 instances and is accessible from
+all of the Availability Zones in the AWS Region where it is created. Learn more about [file systems](https://docs.aws.amazon.com/efs/latest/ug/creating-using.html)
+
+### Create an Amazon EFS file system
+
+A Virtual Private Cloud (VPC) is required where you want EC2 instances to connect to your file system.
```
[the remark this answers is not in the document]
Comment (shivlaks) [reply packages/@aws-cdk/aws-efs/README.md]: yep, simplified the wording.

**Review by eladb — APPROVED** [review]

**Pull request merged**